src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -288,3 +288,48 @@ void WaveformWidgetRenderer::setTrack(TrackPointer track) {
         m_rendererStack[i]->onSetTrack();
     }
 }
+
+WaveformMarkPointer WaveformWidgetRenderer::getCueMarkAtPoint(QPoint point) {
+    WaveformMarkPointer pSelectedMark;
+    const int lineHoverpadding = 5;
+    for (const auto& pMark : m_markLabelOffsets.keys()) {
+        QRectF labelBoundingRectInMarkImageSpace = pMark->m_label.area();
+        int markImageOffsetInWaveformWidgetSpace = m_markLabelOffsets[pMark];
+        int markLineOffset = markImageOffsetInWaveformWidgetSpace + pMark->m_linePosition;
+        QRect labelRectangleInWaveformWidgetSpace;
+        QRect markLineVicinity;
+        if (getOrientation() == Qt::Horizontal) {
+            int labelBoundaryStartOffset =
+                    markImageOffsetInWaveformWidgetSpace +
+                    labelBoundingRectInMarkImageSpace.left();
+            labelRectangleInWaveformWidgetSpace =
+                    QRect(labelBoundaryStartOffset,
+                            labelBoundingRectInMarkImageSpace.top(),
+                            labelBoundingRectInMarkImageSpace.width(),
+                            labelBoundingRectInMarkImageSpace.height());
+            markLineVicinity = QRect(markLineOffset - lineHoverpadding,
+                    0,
+                    lineHoverpadding * 2,
+                    getHeight());
+        } else { /* Vertical */
+            int labelBoundaryStartOffset =
+                    markImageOffsetInWaveformWidgetSpace +
+                    labelBoundingRectInMarkImageSpace.top();
+            labelRectangleInWaveformWidgetSpace =
+                    QRect(labelBoundingRectInMarkImageSpace.left(),
+                            labelBoundaryStartOffset,
+                            labelBoundingRectInMarkImageSpace.width(),
+                            labelBoundingRectInMarkImageSpace.height());
+            markLineVicinity = QRect(0,
+                    markLineOffset - lineHoverpadding,
+                    getWidth(),
+                    lineHoverpadding * 2);
+        }
+
+        if (labelRectangleInWaveformWidgetSpace.contains(point) ||
+                markLineVicinity.contains(point)) {
+            return pMark;
+        }
+    }
+    return static_cast<WaveformMarkPointer>(nullptr);
```

Comment:
```suggestion
    return nullptr;
```
The explicit casting isn't necessary and I don't think it makes the code any easier to understand.